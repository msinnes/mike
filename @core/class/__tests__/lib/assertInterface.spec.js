const assertInterface = require('../../src/lib/assertInterface');

const iAdder = require('../../examples/interfaces/iAdder');
const iNamed = require('../../examples/interfaces/iNamed');

describe('assertInterface', () => {
  it('should be a function', () => {
    expect(assertInterface).toBeDefined();
    expect(assertInterface).toBeInstanceOf(Function);
  });

  it('should not throw an error if the input is a class', () => {
    expect(() => {
      assertInterface(iAdder);
      assertInterface(iNamed);
    }).not.toThrow();
  });

  it('should throw an error if input is not a class', () => {
    expect(() => {
      assertInterface(undefined, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertInterface(null, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertInterface('', 'message');
    }).toThrowErrorMatchingSnapshot();
  });
});
