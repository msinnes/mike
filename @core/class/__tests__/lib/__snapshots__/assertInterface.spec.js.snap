// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertInterface should throw an error if input is not a class 1`] = `"ClassError: message"`;

exports[`assertInterface should throw an error if input is not a class 2`] = `"ClassError: message"`;

exports[`assertInterface should throw an error if input is not a class 3`] = `"ClassError: message"`;
